web3js-ext/test/role_based_key_feepayer.spec.ts

```diff
@@ -0,0 +1,138 @@
+const roleTransactionAccount = web3.eth.accounts.privateKeyToAccount(senderRoleTransactionPriv);
+const roleFeePayerAccount = web3.eth.accounts.privateKeyToAccount(senderRoleFeePayerPriv);
+
+async function checkBalance(address: string) {
+    const balance = await web3.eth.getBalance(address);
+    console.log(`Balance of ${address}: ${web3.utils.fromWei(balance, "ether")} KLAY`);
+    return parseFloat(web3.utils.fromWei(balance, "ether"));
+}
+
+describe("Role-based Key Tests", function () {
+    this.timeout(10000);
+
+    // Before all tests, set up Role-based Key
+    before(async function () {
+        console.log("\n--- Setting Role-based Key ---");
+        const pub1 = getPublicKeyFromPrivate(senderRoleTransactionPriv);
+        const pub2 = getPublicKeyFromPrivate(senderRoleAccountUpdatePriv);
+        const pub3 = getPublicKeyFromPrivate(senderRoleFeePayerPriv);
+
+        const updateTx = {
+            type: TxType.AccountUpdate,
+            from: senderAddr,
+            gasLimit: 100000,
+            key: {
+                type: AccountKeyType.RoleBased,
+                keys: [
+                    { type: AccountKeyType.Public, key: pub1 },
+                    { type: AccountKeyType.Public, key: pub2 },
+                    { type: AccountKeyType.Public, key: pub3 }
+                ]
+            }
+        };
+
+        const signedUpdateTx = await roleAccountUpdate.signTransaction(updateTx);
+        const receipt = await web3.eth.sendSignedTransaction(signedUpdateTx.rawTransaction);
+        console.log("Account Updated:", receipt);
+        assert.isNotNull(receipt.transactionHash, "Account update transaction should succeed");
+    });
+
+    // Test Case 1: Sending a normal transaction with RoleTransaction key
+    it("1. Sending a normal transaction with RoleTransaction key", async function () {
+        const valueTx = {
+            type: TxType.ValueTransfer,
+            from: senderAddr,
+            to: receiverAddr,
+            value: toPeb("0.01"),
+            gasLimit: 100000
+        };
+
+        const signedTx = await roleTransactionAccount.signTransaction(valueTx);
+        const receipt = await web3.eth.sendSignedTransaction(signedTx.rawTransaction);
+        console.log("RoleTransaction signedTx:", receipt.transactionHash);
+        assert.isNotNull(receipt.transactionHash, "RoleTransaction transaction should succeed");
+    });
+
+    // Test Case 2: Attempting to sign a regular transaction with RoleFeePayer key (should fail)
+    it("2. Attempting to sign a regular transaction with RoleFeePayer key (failure test)", async function () {
+        const valueTx = {
+            type: TxType.ValueTransfer,
+            from: senderAddr,
+            to: receiverAddr,
+            value: toPeb("0.01"),
+            gasLimit: 100000
+        };
+
+        try {
+            const signedTx = await roleFeePayerAccount.signTransaction(valueTx);
```

Comment:
It should be signed by a user (from address) first. FeePayer normally sign the tx after the user's sign.
The bug I mentioned before is different. So you are going to encounter another error even if you fix what I mentioned above.

Reply:
https://github.com/kaiachain/kaia-sdk/pull/59#discussion_r1844986026 is followed by this